src/integral.jl

```diff
@@ -76,49 +31,28 @@ function integral(
     _integral(f, geometry, HAdaptiveCubature())
 end
 
-# If algorithm is HAdaptiveCubature, use the generalized n-dim solution
-function integral(
-    f::F,
-    geometry::G,
-    settings::HAdaptiveCubature
-) where {F<:Function, G<:Meshes.Geometry}
-    _integral(f, geometry, settings)
-end
-
-# If algorithm is HAdaptiveCubature, and FP specified, use the generalized n-dim solution
+# with algorithm and T specified
 function integral(
     f::F,
     geometry::G,
-    settings::HAdaptiveCubature,
-    FP::Type{T}
-) where {F<:Function, G<:Meshes.Geometry, T<:AbstractFloat}
+    settings::I,
```

Comment:
```suggestion
    settings::I = HAdaptiveCubature() ,
```
Would it be possible to fuse this method of `integral` with the other one further up by specifying the default for `settings` in the function? Also: You say that the default integration algorithm is `GaussKronrod()` in 1D and `HAdaptiveCubature()` else, but this construct looks like `HAdaptiveCubature` is the default everywhere? Is that correct or can you point me to where the default `GaussKronrod()` is set?

Reply:
Good catch on the defaults. I’ll need to take a closer look at this.

Reply:
I just commit an update that adds conditional default integration rules for the simple two-argument method to address this.

Reply:
Looks good, but I still don't fully get why we need two methods `integral` (one with two arguments and one with four arguments). Wouldn't it be simpler to have only one method with four arguments with defaults for `settings` and `FP`? I updated my suggestion above accordingly.

Reply:
I think this is the way to go but planned to incorporate it into the upcoming breaking change: `FP` becomes a keyword argument, and `rule` can be an optional third argument like this.

Reply:
Sounds good.